capa/features/common.py

```diff
@@ -211,32 +212,29 @@ def evaluate(self, ctx, short_circuit=True):
                 raise ValueError("unexpected feature value type")
 
             if self.value in feature.value:
-                matches[feature.value].extend(locations)
+                matches[feature.value].update(locations)
                 if short_circuit:
                     # we found one matching string, thats sufficient to match.
                     # don't collect other matching strings in this mode.
                     break
 
         if matches:
-            # finalize: defaultdict -> dict
-            # which makes json serialization easier
-            matches = dict(matches)
-
             # collect all locations
             locations = set()
-            for s in matches.keys():
-                matches[s] = list(set(matches[s]))
-                locations.update(matches[s])
+            for locs in matches.values():
+                locations.update(locs)
```

Comment:
this might break a test, im awaiting CI to find out.

we shouldn't be messing with the types here to make serialization work. we should do serialization logic within freeze, not core.